docs/sharing-configurations.md

```diff
@@ -0,0 +1,183 @@
+---
+id: sharing-configurations
+title: Sharing configurations
+---
+
+In case you have many different projects, it can be helpful to have a **shared configuration** which can be used in all of them, instead of copy-pasting the same config for every project.
+
+This page explains how to create, publish and consume a shareable config.
+
+## Creating a Shareable Config
+
+Sharable configs are just [npm packages](https://docs.npmjs.com/about-packages-and-modules#about-packages) that export a single [prettier config file](./configuration.md).
+
+Before we start, make sure you have:
+
+- An account for [npmjs.com](https://www.npmjs.com/) to publish the package
+- Basic understating about [how to create a Node.js module](https://docs.npmjs.com/creating-node-js-modules)
+
+First, create a new package. We recommend creating a [scoped package](https://docs.npmjs.com/cli/v10/using-npm/scope) with the name `@username/prettier-config`.
+
+A minimal package should have at least two files. A `package.json` for the package configuration and an `index.js` which holds the shared prettier configuration object:
+
+```text
+prettier-config/
+├── index.js
+└── package.json
+```
+
+Example `package.json`:
+
+```json
+{
+  "name": "@username/prettier-config",
+  "version": "1.0.0",
+  "description": "My personal Prettier config",
+  "type": "module",
+  "exports": "./index.js",
+  "license": "MIT",
+  "peerDependencies": {
+    "prettier": ">=2.0.0"
+  }
+}
+```
+
+`index.js` is where you put the shared configuration. This file just exports a [regular prettier configuration](./configuration.md) with the same syntax and same options:
+
+```js
+const config = {
+  trailingComma: "es5",
+  tabWidth: 4,
+  singleQuote: true,
+};
+
+export default config;
+```
+
+An example shared configuration repository is available [here](https://github.com/azz/prettier-config).
+
+## Publishing a Shareable Config
+
+Once you are ready, you can [publish your package to npm](https://docs.npmjs.com/creating-and-publishing-scoped-public-packages#publishing-scoped-public-packages):
+
+```sh
+npm publish --access public
+```
+
+## Using a Shareable Config
+
+You first need to install your published configuration, for example:
+
+```sh
+npm install @username/prettier-config --save-dev
+```
```

Comment:
Let's add commands for other clients too.

Reply:
Oops missed link https://github.com/prettier/prettier/blob/44284ba55ca3d4b10b7c5e71463a19dff9c295c7/docs/install.md?plain=1#L8

Reply:
✅